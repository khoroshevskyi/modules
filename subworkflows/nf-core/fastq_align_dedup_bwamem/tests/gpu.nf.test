nextflow_workflow {

    name "Test Subworkflow FASTQ_ALIGN_DEDUP_BWAMEM"
    script "../main.nf"
    workflow "FASTQ_ALIGN_DEDUP_BWAMEM"
    config "../nextflow.config"
    config "./nextflow.config"

    tag "gpu"
    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/fastq_align_dedup_bwamem"
    tag "bwa/mem"
    tag "parabricks/fq2bam"
    tag "samtools/sort"
    tag "samtools/index"
    tag "samtools/flagstat"
    tag "samtools/stats"
    tag "picard/markduplicates"
    tag "samtools/index"
    tag "untar"

    setup {
        run("UNTAR") {
            script "../../../../modules/nf-core/untar/main.nf"
            process {
                """
                input[0] = [
                    [:],
                    file('https://github.com/nf-core/test-datasets/raw/methylseq/reference/Bwameth_Index.tar.gz', checkIfExists: true)
                ]
                """
            }
        }
    }

    test("Params: parabricks/fq2bam single-end | use_gpu") {
        options '-stub'
        when {

            workflow {
                """
                input[0] = Channel.of([
                            [ id:'test', single_end:true ],
                            file('https://github.com/nf-core/test-datasets/raw/methylseq/testdata/SRR389222_sub1.fastq.gz', checkIfExists: true)
                ])
                input[1] = Channel.of([
                            [:],
                            file('https://github.com/nf-core/test-datasets/raw/methylseq/reference/genome.fa', checkIfExists: true)
                ])
                input[2] = Channel.of([
                            [:],
                            file('https://github.com/nf-core/test-datasets/raw/methylseq/reference/genome.fa.fai', checkIfExists: true)
                ])
                input[3] = UNTAR.out.untar
                input[4] = false // skip_deduplication
                input[5] = true // use_gpu
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test("Params: parabricks/fq2bam single-end | use_gpu | skip_deduplication") {
        options '-stub'
        when {
            params {
                skip_deduplication = true
                use_gpu            = true
            }

            workflow {
                """
                input[0] = Channel.of([
                            [ id:'test', single_end:true ],
                            file('https://github.com/nf-core/test-datasets/raw/methylseq/testdata/SRR389222_sub1.fastq.gz', checkIfExists: true)
                ])
                input[1] = Channel.of([
                            [:],
                            file('https://github.com/nf-core/test-datasets/raw/methylseq/reference/genome.fa', checkIfExists: true)
                ])
                input[2] = Channel.of([
                            [:],
                            file('https://github.com/nf-core/test-datasets/raw/methylseq/reference/genome.fa.fai', checkIfExists: true)
                ])
                input[3] = UNTAR.out.untar
                input[4] = true // skip_deduplication
                input[5] = true // use_gpu
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

}
