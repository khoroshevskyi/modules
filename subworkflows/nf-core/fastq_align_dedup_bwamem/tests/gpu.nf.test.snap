{
    "Params: parabricks/fq2bam single-end | use_gpu | skip_deduplication | stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.sorted.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.sorted.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.flagstat:md5,67394650dbae96d1a4fcc70484822159"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.stats:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.idxstats:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    
                ],
                "6": [
                    [
                        "test.flagstat:md5,67394650dbae96d1a4fcc70484822159"
                    ],
                    [
                        "test.idxstats:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        "test.stats:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "7": [
                    "versions.yml:md5,70c24af396804a56c49f53abe6da2d02",
                    "versions.yml:md5,71928d7ca7a14123b4b025bab027d798"
                ],
                "bai": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.sorted.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.sorted.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "multiqc": [
                    [
                        "test.flagstat:md5,67394650dbae96d1a4fcc70484822159"
                    ],
                    [
                        "test.idxstats:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        "test.stats:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "picard_metrics": [
                    
                ],
                "samtools_flagstat": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.flagstat:md5,67394650dbae96d1a4fcc70484822159"
                    ]
                ],
                "samtools_index_stats": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.idxstats:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "samtools_stats": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.stats:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,70c24af396804a56c49f53abe6da2d02",
                    "versions.yml:md5,71928d7ca7a14123b4b025bab027d798"
                ]
            },
            [
                {
                    "FASTQ_ALIGN_DEDUP_BWAMEM:BAM_SORT_STATS_SAMTOOLS:SAMTOOLS_SORT": {
                        "samtools": "1.22.1"
                    }
                },
                {
                    "FASTQ_ALIGN_DEDUP_BWAMEM:PARABRICKS_FQ2BAM": {
                        "pbrun": "4.6.0-1"
                    }
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-20T18:45:42.055034391"
    },
    "Params: parabricks/fq2bam single-end | use_gpu | stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.deduped.sorted.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.deduped.sorted.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.flagstat:md5,67394650dbae96d1a4fcc70484822159"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.stats:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.idxstats:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.deduped.sorted.MarkDuplicates.metrics.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "6": [
                    [
                        "test.deduped.sorted.MarkDuplicates.metrics.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        "test.flagstat:md5,67394650dbae96d1a4fcc70484822159"
                    ],
                    [
                        "test.idxstats:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        "test.stats:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "7": [
                    "versions.yml:md5,6fa0c192669339220d5c5735739188ac",
                    "versions.yml:md5,70c24af396804a56c49f53abe6da2d02",
                    "versions.yml:md5,71928d7ca7a14123b4b025bab027d798",
                    "versions.yml:md5,c8ad9ea35566f100e41588f4c271b663",
                    "versions.yml:md5,e0c30eb274e33f3086ff9e694aae2161"
                ],
                "bai": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.deduped.sorted.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.deduped.sorted.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "multiqc": [
                    [
                        "test.deduped.sorted.MarkDuplicates.metrics.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        "test.flagstat:md5,67394650dbae96d1a4fcc70484822159"
                    ],
                    [
                        "test.idxstats:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        "test.stats:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "picard_metrics": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.deduped.sorted.MarkDuplicates.metrics.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "samtools_flagstat": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.flagstat:md5,67394650dbae96d1a4fcc70484822159"
                    ]
                ],
                "samtools_index_stats": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.idxstats:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "samtools_stats": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.stats:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6fa0c192669339220d5c5735739188ac",
                    "versions.yml:md5,70c24af396804a56c49f53abe6da2d02",
                    "versions.yml:md5,71928d7ca7a14123b4b025bab027d798",
                    "versions.yml:md5,c8ad9ea35566f100e41588f4c271b663",
                    "versions.yml:md5,e0c30eb274e33f3086ff9e694aae2161"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-20T18:45:30.661919995"
    }
}